### PR TITLE
Update Rubocop

### PR DESCRIPTION
**Pull request by jenshenny**
The rubocop target ruby version wasn't updated during the previous ruby bump to 3.0.3. This PR updates the ruby version to 3.0 and fixes violations.

Screen Shot 2022-03-24 at 9 52 16 PM

This PR also updates rubocop gems and regenerates the rubocop todo

### CODE REVIEW

**Comment by sonalkr132** [comment]
If you could update rubocop version that would be awesome. We need to manual update it since it has way too frequent release for dependabot.

**Comment by jenshenny** [comment]
Will do! I've updated a couple of the rubocop gems, I'll take a look at `rubocop-rails` tomorrow since there are some non trivial violations.

**Comment by sonalkr132** [comment]
Thank you. Will check. 
>  if I had time to work on the rubocop todo, are there any cops that you recommend I spend time fixing?

rubocop todos aren't too bad. I would consider having some style guide enforcement on erb files more of a priority.

File: app/models/concerns/rubygem_searchable.rb [hunk 1]
```diff
@@ -46,6 +47,7 @@ def as_indexed_json(_options = {}) # rubocop:disable Metrics/MethodLength
         }
       }.merge!(suggest_json)
     end
+    # rubocop:enable Metrics/CyclomaticComplexity
```
Comment (sonalkr132) [inline app/models/concerns/rubygem_searchable.rb:50]: Why do we need the todo and enable line?

**Review by sonalkr132 — APPROVED** [review]
One minor comment. Otherwise LGTM

File: app/models/concerns/rubygem_searchable.rb (hunk 1 above)
Comment (jenshenny) [reply app/models/concerns/rubygem_searchable.rb]: When I regenerated the rubocop todo file, I ran into this violation and ran `--disable-uncorrectable` to add those comments. Now I realized that the exclude files from `Metrics/CyclomaticComplexity:` was removed for some reason and I manually added it back in.

```
Exclude:
    - 'app/models/concerns/rubygem_searchable.rb'
```

**Comment by simi** [comment]
@jenshenny any chance to rebase this?

**Comment by jenshenny** [comment]
Done! @simi

**Comment by sonalkr132** [comment]
Thank you.

**Pull request merged**